app/redux/wallet/actions.js

```diff
@@ -285,23 +244,36 @@ export const sendTransaction = ({ dstAddress, amount, fee, note }: { dstAddress:
   }
 };
 
-export const addTransaction = ({ tx, accountPK }: { tx: Tx, accountPK?: string }): Action => (dispatch: Dispatch, getState: GetState): Dispatch => {
+export const addTransaction = ({ tx, accountPK }: { tx: Tx, accountPK?: string }): Action => async (dispatch: Dispatch, getState: GetState): Dispatch => {
   try {
     const { accounts, transactions, currentAccountIndex, walletFiles } = getState().wallet;
     const index = accountPK ? accounts.findIndex((account) => account.pk === accountPK) : currentAccountIndex;
     const updatedTransactions = { ...transactions, [index]: [tx, ...transactions[index]] };
-    fileSystemService.updateFile({ fileName: walletFiles[0], fieldName: 'transactions', data: updatedTransactions });
+    await fileSystemService.updateFile({ fileName: walletFiles[0], fieldName: 'transactions', data: updatedTransactions });
+    dispatch(setTransactions({ transactions: updatedTransactions }));
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+export const updateTransactions = ({ txMeta, accountPK }: { txMeta: Tx, accountPK?: string }): Action => async (dispatch: Dispatch, getState: GetState): Dispatch => {
+  try {
+    const { accounts, transactions, currentAccountIndex, walletFiles } = getState().wallet;
+    const index = accountPK ? accounts.findIndex((account) => account.pk === accountPK) : currentAccountIndex;
+    const transactionsArray = transactions[index].map((transaction: Tx) => (transaction.address === txMeta.address ? { ...transaction, ...txMeta } : transaction));
```

Comment:
this will update all transactions having same address. Is this the desired functionality or potential catastrophe?

Reply:
What is the alternative? We are required to update all transactions with the same address.

Reply:
> We are required to update **all** transactions with the same address
is the only and most common usage for update tx function?

Reply:
changed with update all flag.